Makefile.ci.toml

```diff
@@ -0,0 +1,27 @@
+# Format
+[tasks.cargo-fmt]
+category = "LOCAL"
+command = "cargo"
+condition = { channels = ["nightly"] }
+args = ["fmt", "--all", "--check"]
+
+# CI
+[tasks.ci-format]
+category = "CI - CHECK"
+dependencies = ["cargo-fmt"]
+
+[tasks.ci-clippy]
+category = "CI - CHECK"
+command = "cargo"
+args = [
+    "clippy",
+    "--all-targets",
+    "--all-features",
+    "--tests",
+    "--benches",
+    "--examples",
+    "--bins",
+    "--",
+    "-D",
+    "warnings",
+]
```

Comment:
It may be worth (not necessarily now) to go through [clippy checks that are disabled by default](https://rust-lang.github.io/rust-clippy/master/index.html#?levels=allow,none,warn) and enable some of them.

This may introduce a higher rate of false positives. However, it may still be useful to have them flagged up and have the dev `#[allow()]` them in places selectively to acknowledge the issue has been considered and it's deemed to break the lint rule in this particular instance.

It is also possible to run clippy twice. Once with a more permissive setting and apply it to all the code base, and once with stricter settings and apply it only to the main code base (excluding tests, benches, etc. where a more lax approach is often acceptable).

Reply:
Yes, I agree that we should customize the clippy config. I didn't want to have a huge diff on this PR, but I agree that we can introduce more linter if they're acknowledged and relevant.